pimd/pim_iface.c

```diff
@@ -894,8 +894,10 @@ struct in_addr pim_find_primary_addr(struct interface *ifp)
 	 * So let's grab the loopbacks v4 address
 	 * and use that as the primary address
 	 */
-	if (!v4_addrs && v6_addrs && !if_is_loopback(ifp)) {
+	if (!v4_addrs && v6_addrs && !if_is_loopback(ifp)
+	    && !pim_if_is_vrf_device(ifp)) {
 		struct interface *lo_ifp;
+
```

Comment:
this seems like a better solution imo:

```
diff --git a/pimd/pim_iface.c b/pimd/pim_iface.c
index cb31878e0..0572c25c5 100644
--- a/pimd/pim_iface.c
+++ b/pimd/pim_iface.c
@@ -902,7 +902,7 @@ struct in_addr pim_find_primary_addr(struct interface *ifp)
                else
                        lo_ifp = if_lookup_by_name(vrf->name, vrf->vrf_id);
 
-               if (lo_ifp)
+               if (lo_ifp && lo_ifp != ifp)
                        return pim_find_primary_addr(lo_ifp);
        }
 
```

Reply:
could also remove the is a loopback check inthe if statement here too.

Reply:
Yes. Your suggestion looks better. :)

Actual solution internally I had was a function to get any(first) operative loopback interface with an address configured in the vrf and didnt check about vrf device. Instead of working to include vrf device, changed to this as a quick fix.

I will put change suggested by you for now and we will work on a better solution for unnumbered interface later.
Also on other related note, our code assume a interface with certain property as unnumbered, we can discuss on this work towards making it as configuration to avoid unnecessary assumptions and make operations based on it.